src/dolphin/iniFile.ts

```diff
@@ -0,0 +1,278 @@
+    section.setLines(lines);
+  }
+
+  public deleteKey(section_name: string, key: string): boolean {
+    const section = this.getSection(section_name);
+    if (section === undefined) {
+      return false;
+    }
+    return section.delete(key);
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns keys instead of passing it by reference
+   */
+  public getKeys(section_name: string): string[] | boolean {
+    const section = this.getSection(section_name);
+    if (section === undefined) {
+      return false;
+    }
+    return section.keys_order;
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns lines instead of passing it by reference
+   */
+  public getLines(section_name: string, remove_comments = false): string[] {
+    const section = this.getSection(section_name);
+    if (section === undefined) {
+      return [];
+    }
+
+    const lines = section.getLines(remove_comments);
+
+    return lines;
+  }
+
+  public async load(fileName: string, keep_current_data = true): Promise<boolean> {
+    if (!keep_current_data) {
+      this.sections = [];
+    }
+
+    const ins = fs.createReadStream(fileName);
+    ins.on("error", (e) => {
+      console.log("failed to read file with error", e);
+      alert(e);
+    });
+    const rl = readline.createInterface({
+      input: ins,
+      terminal: false,
+    });
+    let current_section = undefined;
+    let first_line = true;
+    for await (let line of rl) {
+      //console.log(line);
+      // Skips the UTF-8 BOM at the start of files. Notepad likes to add this.
+      if (first_line && line.substr(0, 3) === "\xEF\xBB\xBF") {
+        line = line.slice(3);
+      }
+      first_line = false;
+
+      //section line
+      if (line[0] === "[") {
+        //console.log(line, line[0]);
+        const endpos = line.indexOf("]");
+        if (endpos !== -1) {
+          //we have a new section
+          const sub = line.substr(1, endpos - 1);
+          //console.log(sub);
+          current_section = this.getOrCreateSection(sub);
+          //console.log(current_section);
+        }
+      } else {
+        if (current_section !== undefined) {
+          const [key, value] = this.parseLine(line);
+
+          // Lines starting with '$', '*' or '+' are kept verbatim.
+          // Kind of a hack, but the support for raw lines inside an
+          // INI is a hack anyway.
+          if (
+            (key === null && value === null) ||
+            (line.length !== 0 && ["$", "+", "*"].some((val) => line[0] === val))
+          ) {
+            current_section.m_lines.push(line);
+          } else if (key !== null && value !== null) {
+            current_section.set(key, value);
+          }
+        }
+      }
+    }
+
+    return true;
+  }
+
+  public save(filePath: string): boolean {
+    const out = fs.createWriteStream(filePath);
+
+    out.on("error", (e) => {
+      console.log("failed to write file with error", e);
+      alert(e);
```

Comment:
```suggestion
      log.error("failed to write file", e);
```

Again, we probably shouldn't use `alert`.